_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+
+Important details to remember regarding the first version of the Edge Optimized API Gateway:
+* It uses CloudFront distributions, but you *can't* edit the distribution. Adding a WAF, for example, is not possible.
+* It can access other sources, but those sources have to be publicly accessible.
+
+# Regional API Gateways
+In November 2017, Amazon [introduced](https://aws.amazon.com/about-aws/whats-new/2017/11/amazon-api-gateway-supports-regional-api-endpoints/) the **Regional** API Gateway. This variant, as its name implies, is deployed and accessible in a single region. In other words: it's the Edge Optimized API Gateway without the CloudFront distribution.
+
+![Regional](/assets/posts/2018-09-13-API-Gateway-Types-Compared/regional.png)
+
+Benefits to this variant are:
+* You can deploy your own CloudFront distribution in front of your API Gateway. This distribution *can* have a WAF.
+* If your API Gateway is going to be accessed from the same AWS region, the Regional API Gateway will have less latency.
+
+# Integration with Private VPCs
+A big downside to API Gateway has alweays been that any HTTP backend behind the API Gateway needed to be publicly accessible. And because the public IP address of the API Gateway is unknown or unpredictable, IP whitelisting at the backend system was not a viable option. To solve this issue, Amazon [introduced](https://aws.amazon.com/about-aws/whats-new/2017/11/amazon-api-gateway-supports-endpoint-integrations-with-private-vpcs/) integration with private VPCs.
+
+The connection between your API Gateway and your private resources needs two parts:
+* A private Network Load Balancer (NLB) in your VPC in front of the resources you want to access
+* A API Gateway VPC Link that points to that NLB
+
+![VPC Link](/assets/posts/2018-09-13-API-Gateway-Types-Compared/vpc-link.png)
+
+Hurray! With this feature, API Gateway no longer demands your resources to be public and the security of your VPC-based resources is greatly enhanced.
+
+# Private API Gateways
+The integration with private VPC endpoints is an awesome improvement. However, there is still one major issue regarding the security of API Gateway: it is always publicly available. Of course you can implement authentication, rate limiting, API keys, WAF and [resource policies](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-resource-policies.html), but this still feels like a hack when your API Gateway is only used by your VPC-based resources or over AWS VPN. To solve this issue, AWS [introduced](https://aws.amazon.com/blogs/compute/introducing-amazon-api-gateway-private-endpoints/) the **Private** API Gateway in June 2018.
+
+The Private API Gateway is not publicly accessible or resolvable. Instead, it can only be accessed from within your VPC. To achieve this, AWS uses the [PrivateLink](https://aws.amazon.com/about-aws/whats-new/2017/11/introducing-aws-privatelink-for-aws-services/) technology. Diagram time!
```

Comment:
`Diagram time!` -> `Let's look at the diagram again!`

Reply:
I think that's personal preference :D